grails-app/services/au/org/ala/specieslist/UserDetailsService.groovy

```diff
@@ -37,123 +37,59 @@ class UserDetailsService {
 
         try {
             listsWithoutUserIds = SpeciesList.findAllByUserIdIsNullOrUserId("",[max:9999])
-            //listsWithoutUserIds = SpeciesList.findAll("from SpeciesList where userId is null" ,[max:9999])
         } catch (Exception ex) {
             log.error "Failed dynamic finder: ${ex}", ex
         }
 
         if (listsWithoutUserIds.size() > 0) {
-            Map userNamesLookup = authService.getAllUserNameMap()
-            //log.debug "userNamesLookup = ${userNamesLookup}"
-
+            result[TOTAL] = listsWithoutUserIds.size()
             listsWithoutUserIds.each { list ->
-                def user = userNamesLookup.get(list.username?.toLowerCase()) // should be cached after the first call
-
-                if (user && user.userId) {
-                    list.userId = user.userId
-                    log.warn "Saving userid: ${user.userId} (${list.username}) to list id: ${list.id}..."
-
-                    try {
-                        list.save(flush: true)
-                        isSuccessful = true
-                    }
-                    catch (DataIntegrityViolationException e) {
-                        // deal with exception
-                        log.error "Error saving list: ${e}", e
-                    }
-                } else if (list.username =~ /ala\.org/) {
-                    log.error "No user or userId found for username: ${list.username} || user = ${user}"
-                    log.error "Setting username to info@ala.org.au"
-                    list.username = "info@ala.org.au"
-                    list.userId = 2729
-                    try {
-                        list.save(flush: true)
-                        isSuccessful = true
-                    }
-                    catch (DataIntegrityViolationException e) {
-                        // deal with exception
-                        log.error "Error saving list: ${e}", e
+                if (list.username) {
+                    def user = authService.getUserForEmailAddress(list.username, false)
+                    if (user && user.userId) {
+                        list.userId = user.userId
+                        result[VALID] += 1
+                        log.warn "Saving userid: ${user.userId} (${list.username}) to list id: ${list.id}..."
+
+                    } else if (list.username =~ /ala\.org/) {
+                        log.error "No user or userId found for username: ${list.username} || user = ${user}"
+                        log.error "Setting username to info@ala.org.au"
+                        list.username = "info@ala.org.au"
+                        list.userId = 2729
+                        result[DEFAULT] += 1
+                    } else {
+                        log.error "No user or userId found for username: ${list.username} || user = ${user}"
+                        log.error "Setting username to 0"
+                        list.userId = 0
+                        result[INVALID] += 1
```

Comment:
`list.save(flush: true)` is missing from the refactor

Reply:
The Git PR page does not display the changes properly.

list.save() was moved to https://github.com/AtlasOfLivingAustralia/specieslist-webapp/blob/develop/grails-app/services/au/org/ala/specieslist/UserDetailsService.groovy#L145